File: src/main/java/com/rentautosofia/rentacar/entity/Car.java
/*
 * Decompiled with CFR 0_115.
 * 
 * Could not load the following classes:
 *  com.rentautosofia.rentacar.entity.Car
 *  javax.persistence.Column
 *  javax.persistence.Entity
 *  javax.persistence.GeneratedValue
 *  javax.persistence.GenerationType
 *  javax.persistence.Id
 *  javax.persistence.Table
 */
package com.rentautosofia.rentacar.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name="cars")
public class Car {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;
    @Column(nullable=false)
    private String name;
    @Column(nullable=false)
    private String price;
    @Column(nullable=false, length=1000)
    private String imgURL;

    public Car() {
    }

    public Car(String name, String price, String imgURL) {
        this.name = name;
        this.price = price;
        this.imgURL = imgURL;
    }

    public Integer getId() {
        return this.id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPrice() {
        return this.price;
    }

    public void setPrice(String price) {
        this.price = price;
    }

    public String getImgURL() {
        return this.imgURL;
    }

    public void setImgURL(String imgURL) {
        this.imgURL = imgURL;
    }
}

File: src/main/kotlin/com/rentautosofia/rentacar/entity/Car.kt
package com.rentautosofia.rentacar.entity

import com.rentautosofia.rentacar.transmission.Transmission
import javax.persistence.*

@Entity
@Table(name = "cars")
data class Car(@Column var name: String = "",
               @Column var price: Int = 0,
               @Column(name = "imgurl", length = 1000) var imgUrl: String = "",
               @Column var LPG: Boolean = false,
               @Column @Enumerated(EnumType.STRING)
               var transmission: Transmission = Transmission.MANUAL,
               @Column var seatCount: Int? = null,
               @Id
               @GeneratedValue(strategy = GenerationType.IDENTITY)
               var id: Int = 0,
               @Column var deposit: Int = 0) {

    /**
     * @param days - for backwards compatibility
     * The price was supposed to change for a different
     * rental duration
     */
    fun getPricePerDayFor(days: Int) = this.price //when (days) {
//        in 1 .. 3 -> this.price
//        in 4 .. 8 -> this.price - 7
//        in 9 .. 15 -> this.price - 15
//        in 16 .. Int.MAX_VALUE -> this.price - 20
//        else -> throw IllegalArgumentException("Cannot get get price per day for $days (NON-POSITIVE) days!!")
//    }
}
//
//fun car(function: Car.() -> Unit): Car {
//    val car = Car()
//    car.function()
//    return car
//}
